### ISSUE TITLE
[ImportMap] Resolution of 'scopes' is affect by 'imports' during extending import map

### ISSUE THREAD

**Issue by shrinktofit** [systemjs/systemjs#2281]
### Demonstration

Currently when merging an import map. SystemJS do:
1. `resolveAndComposePackages` the `imports` field of the new import map;
2. `resolveAndComposePackages` each specifier map of the `scopes` field of the new import map.

`resolveAndComposePackages` will resolve the specifier map according to the "current" import map. But every `resolveAndComposePackages` call would update the "current" import map. As a result, resolution of `scope_2` is affect by `scope_1` and `scope_0` and `imports`. This behavior is not mentioned in import map proposal.

For example:
```json5
{
  "imports": {
     "/foo/": "/bar/"
  },
  "scopes": {
    "/": {
      "bar": "/foo/index.js"
    }
  }
}
```

`System.import('bar')` at root URL should import `/foo/index.js` instead of `/bar/index.js`.

**Comment by joeldenning** [comment]
Thanks @shrinktofit for the description. This is related to [cascading imports](https://github.com/WICG/import-maps/issues/137), which is not part of the import maps v1 spec but is implemented in SystemJS. The exact desired behavior of cascading imports is not totally clear to me, and may not have ever been fully defined.

I've created https://codesandbox.io/s/restless-butterfly-z3k0w?file=/index.html which shows the behavior you described.

It seems that you're proposing that cascades should apply only to specifiers, not to resolved URLs. I'm not sure whether that fits the proposal in the import maps spec. @guybedford can you comment here?

**Comment by guybedford** [comment]
Yes this sounds like a bug in the cascading implementation that it is applying to URL-like specifiers and not just bare specifiers.
